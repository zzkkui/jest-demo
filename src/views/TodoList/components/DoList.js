import React from 'react'

const DoList = (props) => {
  const { list } = props;
  return (
    <div className="do-list">
      <div className="do-list-title">
        已经完成
        <div className="do-list-count" data-test="count">{list.length}</div>
      </div>
      <ul className="do-list-ul">
        {
          list.map((n, i) =>
            n.isChecked ? <li className="do-list-item" data-test="list-item" key={i}>
              <span>{n.value}</span>
            </li> : '')
        }
      </ul>
    </div>
  )
}

export default DoList
