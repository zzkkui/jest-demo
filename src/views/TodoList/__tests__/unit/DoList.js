
import React from 'react';
import { shallow } from 'enzyme';
import DoList from '../../components/DoList';
// import { findTestWrapper } from '../../../../utils/testUtils'

const listData = [{
  value: 'jest',
  isFocus: true,
  isChecked: true
}, {
  value: 'TDD',
  isFocus: true,
  isChecked: true
}, {
  value: 'react',
  isFocus: false,
  isChecked: false
}]

describe('DoList 组件', () => {

  it('样式渲染正常', () => {
    const warpper = shallow(<DoList list={listData} />)
    expect(warpper).toMatchSnapshot()
  })

  it('数据为空时样式渲染正常', () => {
    const warpper = shallow(<DoList list={[]} />)
    expect(warpper).toMatchSnapshot()
  })
});


