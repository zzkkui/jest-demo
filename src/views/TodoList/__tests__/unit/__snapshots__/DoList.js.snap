// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoList 组件 数据为空时样式渲染正常 1`] = `
<div
  className="do-list"
>
  <div
    className="do-list-title"
  >
    已经完成
    <div
      className="do-list-count"
      data-test="count"
    >
      0
    </div>
  </div>
  <ul
    className="do-list-ul"
  />
</div>
`;

exports[`DoList 组件 样式渲染正常 1`] = `
<div
  className="do-list"
>
  <div
    className="do-list-title"
  >
    已经完成
    <div
      className="do-list-count"
      data-test="count"
    >
      3
    </div>
  </div>
  <ul
    className="do-list-ul"
  >
    <li
      className="do-list-item"
      data-test="list-item"
      key="0"
    >
      <span>
        jest
      </span>
    </li>
    <li
      className="do-list-item"
      data-test="list-item"
      key="1"
    >
      <span>
        TDD
      </span>
    </li>
  </ul>
</div>
`;
